File: src/tutorial_py_pkg/tutorial_py_pkg/topic_publisher.py
import rclpy
from rclpy.node import Node
from tutorial_cpp_pkg.msg import TestTopic

class MyNode(Node):
    def __init__(self):
        super().__init__("Python_Topic_Publisher")
        self.declare_parameter(name="param_int64", value=0)
        self.declare_parameter(name="param_bool", value=True)
        self.declare_parameter(name="param_string", value="PYTHON")
        self.publisher_ = self.create_publisher(TestTopic, "py_tutorial_topic", 10)
        self.timer_ = self.create_timer(1.0, self.publish_topic)
        self.get_logger().info("Start publish topic...")
        
    def publish_topic(self):
        msg = TestTopic()
        msg.msg_int64 = self.get_parameter("param_int64").value
        msg.msg_bool = self.get_parameter("param_bool").value
        msg.msg_string = self.get_parameter("param_string").value
        self.publisher_.publish(msg)
        self.get_logger().info("Topic published!")

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    rclpy.shutdown()
    
if __name__ == "__main__":
    main()

File: src/tutorial_cpp_pkg/src/topic_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "tutorial_cpp_pkg/msg/test_topic.hpp"

class MyNode: public rclcpp::Node
{
public:
    MyNode() : Node("CPP_Topic_Publisher") {
        RCLCPP_INFO(this->get_logger(), "Cpp Topic Publisher");
        pub_ = this->create_publisher<tutorial_cpp_pkg::msg::TestTopic>("cpp_tutorial_topic", 10);
        timer_ = this->create_wall_timer(
            std::chrono::seconds(1),
            std::bind(&MyNode::publishTopic, this)
            );
        RCLCPP_INFO(this->get_logger(), "Start publish topic...");
    }

private:
    void publishTopic() {
        auto msg = tutorial_cpp_pkg::msg::TestTopic();
        msg.msg_int64 = 111;
        msg.msg_bool = true;
        msg.msg_string = "CPP";
        pub_->publish(msg);
        RCLCPP_INFO(this->get_logger(), "Topic published");
    }

private:
    rclcpp::Publisher<tutorial_cpp_pkg::msg::TestTopic>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}